andela/Ah-backend-guardians
# #162685666 Users can comment on articles

#### What does this PR do?
This PR enables only authenticated users to comment on an article(s).


#### Description of Task to be completed?
Authenticated users are able to comment on articles that have been published. The users are able to perform all the CRUD functionality on the comments of the articles.
Available endpoints include
```
GET api/<slug>/comments/         Gets all comments for a particular article
```
```
POST api/<slug>/comments/         Creates a comment for a particular article
```
```
GET api/<slug>/comments/<id>/         Get a specific comment of an article
```
```
PUT api/<slug>/comments/<id>/         Update a specific comment of an article
```
```
DELETE api/<slug>/comments/<id>/         Delete a specific comment of an article
```
```
POST api/<slug>/comments/<id>/thread/         Comment on a comment
```
```
GET api/<slug>/comments/<id>/thread/         Get entire thread of a comment
```


#### How should this be manually tested?
Run the following commands in your terminal.

- `git clone https://github.com/andela/Ah-backend-guardians.git`
- `cd Ah-backend-guardians`
- `virtualenv venv`
- `source venv/bin/activate`
- `git checkout ft-comment-on-articles-162685666`
- `python3 manage.py runserver`


Alternatively, you can test the app using the Heroku link below
https://ah-backend-guardians-sta-pr-25.herokuapp.com/

Next open POSTMAN and begin to test out the end points listed above.
The data required in a POST and PUT is in the form
```
{
	"body":  ""
}
```

Note: You must be logged in to access the endpoints and atleast one article should be present in your database.


#### What are the relevant pivotal tracker stories?
#162685666


#### Screenshots (if appropriate)
This shows a GET all comments for an article
![image](https://user-images.githubusercontent.com/26769886/51371844-5d46f680-1b0c-11e9-8539-da61a6f7ca3f.png)
